app/models/goal.py

```diff
@@ -1,5 +1,24 @@
 from app import db
-
+from app.models.task import Task
 
 class Goal(db.Model):
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal", lazy=True)
+
+
+    def to_dict(self):
+        goal_dict = {
+            "id": self.goal_id,
+            "title": self.title
+        }
+
+        if self.tasks:
+            goal_dict["tasks"] = [task.to_dict() for task in self.tasks]
+
+        return goal_dict
+
+    def from_dict(goal_data):
+        return Goal(
+            title=goal_data["title"]
+        )
```

Comment:
I recommend adding the `@staticmethod` decorator here.  Without the decorator that designates this function as a static method, you can call it from an instance of Goal, and Python will pass in `self` as the `goal_data` and if you pass in something for goal_data it will throw an error that it has received more positional arguments than the function accepts.  With the `@staticmethod` decorator, you can call this function from the class or an instance, and it will behave the same. 
```suggestion
    @staticmethod
    def from_dict(goal_data):
        return Goal(
            title=goal_data["title"]
        )
```